salt/metalk8s/macro.sls

```diff
@@ -2,12 +2,16 @@
 
 {%- macro pkg_installed(name='') -%}
   {%- set package = repo.packages[name] | default({}) %}
+  {%- set version = package.version | default(None) %}
+  {%- set deps_list = salt.metalk8s_package_manager.list_pkg_deps(name, version) %}
```

Comment:
You can't do it in jinja.
As when the jinja is rendered, the repository may not have been configured yet.